AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +3814,265 @@ var interpolate = function(fromValue, toValue, progress)
 
 @end
 
+@implementation CPWindow (TrackingAreaAdditions)
+
+- (void)_addTrackingAreaView:(CPView)aView
+{
+    var trackingAreas = [aView trackingAreas];
+    
+    for (var i = 0; i < trackingAreas.length; i++)
+        [self _addTrackingArea:trackingAreas[i]];
+}
+
+- (void)_removeTrackingAreaView:(CPView)aView
+{
+    var trackingAreas = [aView trackingAreas];
+    
+    for (var i = 0; i < trackingAreas.length; i++)
+        [self _removeTrackingArea:trackingAreas[i]];
+}
+
+- (void)_addTrackingArea:(CPTrackingArea)trackingArea
+{
+    var trackingAreaView = [trackingArea _referencingView];
+    
+    if (![_trackingAreaViews containsObject:trackingAreaView])
+        [_trackingAreaViews addObject:trackingAreaView];
+    
+    // If CPTrackingAssumeInside option is set, put the tracking area in the _mouseEnteredStack
+    
+    if ([trackingArea options] & CPTrackingAssumeInside)
+        [_mouseEnteredStack addObject:trackingArea];
+}
+
+- (void)_removeTrackingArea:(CPTrackingArea)trackingArea
+{
+    // If mouse is in the tracking area, we remove it from the stack to avoid to fire a future mouseExited event
+    
+    if ([_mouseEnteredStack containsObject:trackingArea])
+        [_mouseEnteredStack removeObject:trackingArea];
```

Comment:
Is it -removeObject: or removeObjectIdenticalTo: ? The first one means "remove object equal to".
Also you don't need to check with containsObject(IdenticalTo?) because the check is already included in removeObject(IdenticalTo?):